### ISSUE TITLE
 🔥 getToken() and Android with low Network

### ISSUE THREAD

**Issue by tgensol** [invertase/react-native-firebase#1819]
Hi there,

## Issue

My UI was not responding, I could scroll but any tap will not work until an certain amount of time, and then I was getting all previous tapping (looks like the react-navigation was not responding until something). This was only happening on **Android** with a low or no network (My app was not working well in the subway)

After some digging, I found that these line was causing the issue : 

> firebase.messaging().getToken()

If I do not have it, the UI is responding well with or without network. If I do have it, the UI seems to be waiting for the function to resolve.

On iOS, it looks like the function is resolving a token without or without network, so I do not have the issue,

I am calling the function in the first tab of my react-navigation bottom tabs,

Did someone hear about something like this ? Any idea how to solve it ?

## Environment

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [X] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both

- **Build Tools:**
  - `ADD_SOMETHING_HERE` e.g. Xcode 10, Android Studio 3.2
- **`React Native` version:**
  - 0.57.8
- **`React Native Firebase` library version:**
  - 5.1.1 and 5.2.0, but I did not try before
- **`Firebase` module(s) you're using that has the issue:**
  - [ ] **N/A**
  - [ ] Authentication
  - [ ] Analytics
  - [ ] Cloud **Firestore**
  - [X] Cloud **Messaging** (FCM)
  - [ ] Crashlytics
  - [ ] Dynamic **Links**
  - [ ] **Functions** Callable
  - [ ] Invites
  - [ ] Instance ID
  - [ ] Notifications
  - [ ] Performance Monitoring
  - [ ] Realtime **Database**
  - [ ] Remote **Config**
  - [ ] Storage
- **Are you using `TypeScript`?**
  - [ ] No
  - [X] Yes, version: `N/A`
- **Are you using Expo, e.g. `ExpoKit`?**
  - [x] No
  - [ ] Yes, I've _not_ ejected
  - [ ] Yes, but I **have** ejected to `ExpoKit`
  - [ ] Yes, but I **have** ejected to vanilla React Native
  - Expo version: `N/A`

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](./../../CONTRIBUTING.md)

**Comment by nmchr7** [comment]
Hi,

I've experienced similar issue on Android. I'm also using react-navigation with tabs and used to call `firebase.messaging().getToken()` in `componentDidMount`. I'm guessing that is what you are doing now.

I just solved it today by handling this using redux-saga. If you're already using sagas this might be a good way to go. I think the problem is that you're handling a side effect in the component that is responsible for rendering your UI.

Hope it helps,
Cheers
